docs: add note about porting guide to RELEASING.md (#39)
ansible-collections/partner-certification-checker: RELEASING.md

To track changes in this repository made between versions and to generate a changelog, we use [changelog fragments](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment).

### When updating tools versions in the reusable workflow

> When updating tools versions in the reusable workflow, make sure the changelog and notifications to partners to update the workflow version in their repos reflect porting guides for related breaking changes.

Reviewer: ```suggestion
When updating versions of tools in the reusable workflow, ensure that the changelog, and any notifications to partners, include porting guides for related breaking changes.
```

> For example, when elevating the `ansible-core` version, say, from `2.16` to `2.17`, reflect it in a corresponding changelog fragment, noting that because workflow version N runs the `ansible-test sanity` command from ansible-core 2.17, if a partner has `tests/sanity/ignore-2.16.txt` file, they need to copy it to `tests/sanity/ignore-2.17.txt` to prevent errors.

Reviewer: ```suggestion
For example, when bumping the `ansible-core` version from `2.16` to `2.17`, create a corresponding changelog fragment. It should note that, because workflow version N runs the `ansible-test sanity` command from ansible-core 2.17, if a partner has a `tests/sanity/ignore-2.16.txt` file, they need to copy it to `tests/sanity/ignore-2.17.txt` to prevent errors.
```
